### ISSUE TITLE
Add support for themes

### ISSUE THREAD

**Issue by edreamleo** [leo-editor/leo-editor#788]
Per recent discussions:

- [x] Allow redirection of "hard" tk/solarized colors to theme-specific colors.
   This will be done in LeoQtLog.put
- [x] Add "error", "info" and "warning" to the list of redirectable colors.
- [x] Not needed. Change color="red" to color="error" in various calls to g.es, etc. in Leo's core.
- [x] Terry reports that ​until recent changes, these settings worked:

- (fixed) @color log_text_foreground_color = #93a1a1
- (fixed) @color log_text_background_color = #073642
- (fixed) @color log_error_color = #dc322f
- (fixed) @color log_warning_color = #268bd2

**Comment by tbnorth** [comment]
Maybe also support type='error' etc., for maximum readability.

**Comment by edreamleo** [comment]
Rev de0dcfa completes this item. Leo already had substantial capabilities in this area.

The new code updates g.actualColor so it uses log_xxx_color settings if available.  LeoQtLog.put contains some fallback code. The following script demonstrates that the new code makes everything legible, with light and dark themes, or with no theme:
```python
table = (
    # non-solarized names...
    'error', 'info', 'warning', 'black', 'white', 
    # A completely bogus color name should still be legible.
    'xxx',
    # Tk colors...
    'firebrick', 'grey', 'pink', 'purple', 
    # Solarized colors...
    'blue', 'cyan', 'green','magenta', 'orange', 'red', 'violet', 'yellow',
)
for s in table:
    g.es(s, color=s)
```

**Comment by edreamleo** [comment]
I am going to reopen this item handle the last item in the first comment.

**Comment by edreamleo** [comment]
Fixed at acd8f99
